### PR TITLE
Fix padding for min/max when input is float32

### LINKED ISSUE #22902: ttnn.fill_implicit_tile_padding does not place the correct values when input is float32
## What is the behavior
`ttnn.fill_implicit_tile_padding` does not place the correct values in padded regions of a tile when input dtype is float32.

## Code Snippet
```python
import random
import torch
import ttnn

# Parameters
random.seed(0)
tensor_shape = [2, 2]

# Setup
device = ttnn.open_device(device_id=0)

torch_tensor = torch.tensor([-3.0, -5.0, -40, -9.0], dtype=torch.float32).reshape(tensor_shape)
ttnn_tensor = ttnn.from_torch(torch_tensor, layout=ttnn.TILE_LAYOUT, device=device)
ttnn_tensor = ttnn.fill_implicit_tile_padding(ttnn_tensor, fill_value=2)
ttnn_tensor = ttnn.reshape(ttnn_tensor, (32, 32), (32, 32))

ttnn_result_host = ttnn.from_device(ttnn_tensor)
ttnn_result = ttnn.to_torch(ttnn_result_host)
print(ttnn_result)

# Cleanup
ttnn.close_device(device)
```

## Why this matters
Reduction family of ops call implicit padding to pad +/-infinity as required for min/max. These produce incorrect results as a result of this bug.

### PR DESCRIPTION
**Pull request by nsorabaTT**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/19480

### Problem description
When input has padded fields, min/max incorrectly returns 0 as output.

### What's changed
Min/Max currently use fill_implicit_padding to place +/-inf when there are padded locations. This PR fixes an issue with the padding for float32 inputs. It also specifically adds bfloat16 and float32 dtype tests to sweeps.

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passing https://github.com/tenstorrent/tt-metal/actions/runs/15427415348
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI fails, but matches main: https://github.com/tenstorrent/tt-metal/actions/runs/15425183923
- [x] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI Passing https://github.com/tenstorrent/tt-metal/actions/runs/15434472171
- [x] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI Passing: https://github.com/tenstorrent/tt-metal/actions/runs/15425188872
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [x] New/Existing tests provide coverage for changes

Hat tip to @vsureshTT for finding the fix

### CODE REVIEW

**Review by bbradelTT — COMMENTED** [review]
This PR needs a unit test for fill_pad that fails without this change and passes with this change.

File: tests/ttnn/unit_tests/operations/test_fill_pad.py [hunk 1]
```diff
@@ -98,6 +99,8 @@ def test_fill_pad_bfloat16(
     output_tensor = ttnn.fill_implicit_tile_padding(input_tensor, fill_value, memory_config=output_mem_config)
     padded_torch_output_tensor = ttnn.from_device(output_tensor).to_torch_with_padded_shape()
 
+    print(padded_torch_tensor)
```
Comment (bbradelTT) [inline tests/ttnn/unit_tests/operations/test_fill_pad.py]: assert_with_pcc should print the tensors if there's an error. Please remove these prints to avoid making CI logs even longer.

**Review by bbradelTT — APPROVED** [review]

File: tests/ttnn/unit_tests/operations/test_fill_pad.py (hunk 1 above)
Comment (nsorabaTT) [reply tests/ttnn/unit_tests/operations/test_fill_pad.py]: Yes, fixed.

**Comment by nsorabaTT** [comment]
> This PR needs a unit test for fill_pad that fails without this change and passes with this change.

Yes, added a unit test to cover float32 inputs. Failing before this PR, passing now.

**Review by vsureshTT — APPROVED** [review]
Looks good to me

**Comment by amorrisonTT** [comment]
This looks just fine however on [here](https://docs.google.com/spreadsheets/d/1qCQINfEkIDIF9BXDvsuqwQQJkKhK0mViTFVBLklejZM/edit?gid=0#gid=0) it is claimed that fill pad does not support float32. What do you think @yugi957 ?

**Review by jvegaTT — APPROVED** [review]
